samples/bluetooth/mesh/light/src/smp_dfu.c

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/init.h>
+
+#include <img_mgmt/img_mgmt.h>
+#include <os_mgmt/os_mgmt.h>
+#include <zephyr/mgmt/mcumgr/smp_bt.h>
+
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+
+static struct k_work advertise_work;
+
+static const struct bt_data ad[] = {
+	BT_DATA_BYTES(BT_DATA_FLAGS, (BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR)),
+	BT_DATA_BYTES(BT_DATA_UUID128_ALL, 0x84, 0xaa, 0x60, 0x74, 0x52, 0x8a, 0x8b, 0x86, 0xd3,
+		      0x4c, 0xb7, 0x1d, 0x1d, 0xdc, 0x53, 0x8d),
+};
+
+static void advertise(struct k_work *work)
+{
+	int rc;
+
+	rc = bt_le_adv_start(BT_LE_ADV_CONN_NAME, ad, ARRAY_SIZE(ad), NULL, 0);
+	if (rc) {
+		printk("SMP BT service advertising failed to start (rc %d)\n", rc);
+		return;
+	}
+
+	printk("SMP BT service advertising successfully started\n");
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	k_work_submit(&advertise_work);
```

Comment:
A false schedule may happen if disconnect happens for Mesh Provisioning or Mesh Proxy:
```
Bluetooth initialized
Mesh initialized
SMP BT service advertising successfully started
I: Primary Element: 0x0002
W: opcode 0x2039 status 0x09
SMP BT service advertising failed to start (rc -120)
```
The simplest fix I think would be to check error code of `bt_le_adv_start` against `-EALREADY`. But maybe the more correct way would be to use extended advertising API here and handle this somehow using `struct bt_le_ext_adv_cb` callbacks.

Reply:
So I guess that is an issue in our other sample using this as well then?

Reply:
Yes

Reply:
> A false schedule may happen if disconnect happens for Mesh Provisioning or Mesh Proxy:
> 
> ```
> Bluetooth initialized
> Mesh initialized
> SMP BT service advertising successfully started
> I: Primary Element: 0x0002
> W: opcode 0x2039 status 0x09
> SMP BT service advertising failed to start (rc -120)
> ```
> 
> The simplest fix I think would be to check error code of `bt_le_adv_start` against `-EALREADY`. But maybe the more correct way would be to use extended advertising API here and handle this somehow using `struct bt_le_ext_adv_cb` callbacks.

I've tried and successfully added the extended advertiser to smp_dfu.c. Not sure how this can be used to avoid redundant advertisment of smp though. As far as I can see the `struct bt_le_ext_adv_cb` offers a `sent`, `connected` and `scanned` cb. At least the `connected` CB triggers for both device manager app and for the prov/proxy connections in the mesh app.

Reply:
Maybe the 'sent` callback can be used 🤔

Reply:
I have made an attempt using the `sent` CB. It seems to work Ok in testing. Review it and see what you think about it